File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.py
N, M = map(int, input().split())

basket = []
temp = []

for i in range(N):
    basket.append(i+1)

for _ in range(M):
    i, j = map(int, input().split())
    temp = basket[:]
    for k in range(i-1, j):
        basket[k] = temp[j - 1 - k + i - 1]
    temp = basket[:]

print(*basket)

File: 백준/Bronze/10811. 바구니 뒤집기/바구니 뒤집기.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		Scanner sc = new Scanner(System.in);

		StringTokenizer st = new StringTokenizer(sc.nextLine());

		int N = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());

		int[] basket = new int[N];
		int[] temp;

		for (int i = 0; i < N; i++) {
			basket[i] = i + 1;
		}

		for (int k = 0; k < M; k++) {
			st = new StringTokenizer(sc.nextLine());
			int i = Integer.parseInt(st.nextToken());
			int j = Integer.parseInt(st.nextToken());

			temp = basket.clone();

			for (int l = i - 1; l < j; l++) {
				basket[l] = temp[j - 1 - (l - i + 1)];
			}

			temp = basket.clone();
		}

		sc.close();

		for (int number : basket) {
			System.out.print(number + " ");
		}

	}
}